src/audio/codec_adapter/codec/generic.c

```diff
@@ -339,7 +339,7 @@ int codec_reset(struct comp_dev *dev)
 	/* Codec reset itself to the initial condition after prepare()
 	 * so let's change its state to reflect that.
 	 */
-	codec->state = CODEC_INITIALIZED;
+	codec->state = CODEC_IDLE;
```

Comment:
Not following, if we reset() the codec then any configuration via IPC will also be reset like channels, rate, format etc ?

Reply:
Yes it will be reset to their default values.